operators/endpointmetrics/pkg/rendering/renderer.go

```diff
@@ -56,6 +56,13 @@ func Render(
 	hubInfo *operatorconfig.HubInfo,
 ) ([]*unstructured.Unstructured, error) {
 
+	isKindTest := false
+	if strings.Contains(hubInfo.ClusterName, "kind") {
+		//remove -kind from the cluster name
```

Comment:
This is because the only way to detect whether we run in kind is via the cluster name? So i.e add the suffix in the `observability_addon` controller and remove it again here?

This kind of name mangling doesn't seem super nice to me.

It seems the `isHubMetricsCollector` and `installPrometheus` are determined via the environment, could we infer this information directly form the environment instead here, or alternatively pass a `isKindTest` as an argument to Render() ?

Reply:
Makes sense. I've added checking for the environment var in render instead.

Reply:
Thanks, that looks much cleaner to me :+1: